### PR TITLE
Adds an introduction to the Pipelines SDK

### PR DESCRIPTION
**Pull request by sarahmaddox**
Added an overview of the Pipelines SDK and the various ways you can use it to build and interact with pipelines.

Replaced the [DSL overview](https://www.kubeflow.org/docs/pipelines/sdk/dsl-overview/) to give the full picture of authoring pipelines with SDK and DSL combined.

Moved the content about pipeline parameters into a new file: /docs/pipelines/sdk/parameters.md 

Fixes https://github.com/kubeflow/website/issues/809
Fixes https://github.com/kubeflow/website/issues/810

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/website/811)

### CODE REVIEW

**Comment by sarahmaddox** [comment]
Preview: https://deploy-preview-811--competent-brattain-de2d6d.netlify.com/docs/pipelines/sdk/sdk-overview/

**Comment by sarahmaddox** [comment]
Removing reviewers as this PR will be noisy until it's ready for review.

/uncc @jlewi 
/uncc @kunmingg

**Comment by sarahmaddox** [comment]
This PR is now ready for review. I'm adding a hold to allow for multiple reviewers.
/hold

**Comment by sarahmaddox** [comment]
/assign @gaoning777 
/cc @zanedurante

Preview: 

* New guide: https://deploy-preview-811--competent-brattain-de2d6d.netlify.com/docs/pipelines/sdk/sdk-overview/
* Content extracted from the now-deleted DSL overview: https://deploy-preview-811--competent-brattain-de2d6d.netlify.com/docs/pipelines/sdk/parameters/

**Comment by zanedurante** [comment]
Just read the intro to the SDK -- looks great!

File: content/docs/pipelines/sdk/sdk-overview.md [hunk 1]
```diff
@@ -0,0 +1,532 @@
...
+  SDK.
+
+1. Write a component function using the Kubeflow Pipelines DSL to define your
+  pipeline's interactions with the component’s Docker container. Your
+  component function must return a
+  [`kfp.dsl.ContainerOp`](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.dsl.html#kfp.dsl.ContainerOp).
+  Optionally, you can use the [`kfp.dsl.component` 
+  decorator](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.dsl.html#kfp.dsl.component)
+  to enable [static type checking](/docs/pipelines/sdk/static-type-checking) in 
+  the DSL compiler. To use the decorator, you can add the `@kfp.dsl.component`
```
[the remark this answers is not in the document]
Comment (sarahmaddox) [reply content/docs/pipelines/sdk/sdk-overview.md]: When writing the doc, I went back and forth on this one. The reason I included "you can" is that, AFAIK, using the annotation is a shortcut. ("Syntactic sugar.") People can choose to call the decorator function instead. I didn't want to get into explaining how to use decorators, so I thought that including "you can" would signal to those in the know that the annotation is one option, without adding too much detail to scare off the less experienced programmers.

What do you think - is there a better way of explaining it?

File: content/docs/pipelines/sdk/sdk-overview.md [hunk 2]
```diff
@@ -0,0 +1,532 @@
...
+    URI. In the parameters, you provide the Python function that does the work 
+    of the pipeline component, a Docker image to use as a base image, 
+    and the URI for the target image (for example, a container registry).
+
+* [`kfp.components`](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.components.html)
+  includes classes and methods for interacting with pipeline components. 
+  Methods in this package include, but are not limited to, the following:
+
+  * `kfp.components.func_to_container_op` converts a Python function to a 
+    pipeline component and returns factory function.
```
[the remark this answers is not in the document]
Comment (sarahmaddox) [reply content/docs/pipelines/sdk/sdk-overview.md]: Done.

**Comment by sarahmaddox** [comment]
> Just read the intro to the SDK -- looks great!

Thanks @zanedurante!

**Pull request merged**